pkg/middlewares/auth/digest_auth.go

```diff
@@ -63,6 +63,19 @@ func (d *digestAuth) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 
 	username, authinfo := d.auth.CheckAuth(req)
 	if username == "" {
+		headerField := d.headerField
+		if d.headerField == "" {
+			headerField = "Authorization"
+		}
+
+		auth := goauth.DigestAuthParams(req.Header.Get(headerField))
```

Comment:
similarly, it is a bit silly that we are redoing the computation for the username here, but d.auth.CheckAuth is more complicated in the digest auth case than in the basic auth case, so it does not bother me as much here as in the basic auth case.

Reply:
Let say that if you agree to don’t change CheckAuth signature, this one would remain as it is ?

Reply:
I mean, maybe changing CheckAuth signature is no big deal, if we can easily change the other calls everywhere else. Or we could also write another method that does the same thing as CheckAuth but returns the username. (and CheckAuth would just be a wrapper of it, that drops the username before returning.). (And same thing for BasicAuth).
But yes, if it does not bother you, feel free to ignore that comment.